Add basic test suite for core MEDFORD functionality

alexisylee opened the pull request:
This PR adds a comprehensive test suite to validate basic MEDFORD functionality before implementing bagit integration or the validator component.

Included:
- Test cases for all core MEDFORD components (contributor, data, date, expedition, file, funding, journal, keyword, method, paper, software, species)
- Input .mfd files for testing various scenarios
- Expected output JSON files for validation
- Test runner script and setup utilities

This test suite was implemented after fixing the JSON output functionality and serves as a foundation for testing core features before moving to more advanced components like bagit or validation. Run the test suite with `./testsuite/setup.sh` to verify all basic functionality works as expected.

Commit:
testsuite for basic functionalities

Commit:
testsuite done

alvacouch commented on src/MEDFORD/objs/linecollections.py:
```diff
@@ -268,6 +268,9 @@ def __init__(self, headline: NovelDetailLine, extralines: Optional[List[Continue
             self.is_header = True
         else :
             self.is_header = False
+        print(f"\nInside Detail's init", file=sys.stderr)
```
What I do with these is to put them in if __DEBUG__: clauses, so that they don't appear all the time. 
Then we can either set
__DEBUG__=True
or 
__DEBUG_=False
to make them appear or disappear. 

For extra added "special sauce," note that we can program the test environment to set __DEBUG__=True before it runs.

alvacouch commented on src/MEDFORD/objs/lines.py:
```diff
@@ -220,6 +221,9 @@ class Line() :
     def __init__(self, lineno : int, line : str) :
         self.lineno = lineno
         self.line = line
+        print(f"/n Inside Line's init")
```
Same comment here. These are debugging writes.  Put in a __DEBUG__ clause.

alexisylee committed:
formatted code and added debug clauses

infispiel committed:
Merge branch 'dev' into dev_basic_testsuite

infispiel committed:
Adjust setup.sh to work with MEDFORD module

Previous implementation:
1. treated MEDFORD like a script (which it was)
2. used a hard link to MEDFORD, which doesn't work on other people's machines

infispiel committed:
Propogate write_json parameter

Tests dependended on presence of --write_json param, but param was
never actually passed to the MEDFORD parser. This has been fixed.

infispiel committed:
Adjust usability of python testsuite script
1. Adjusted testsuite_medford.py to treat MEDFORD like a module
2. Adjusted " to use the same python interpreter that was used to run it.
(e.g. in my case, I use `uv run python3 testsuite_medford.py`; this makes sure
it uses the venv uv is maintaining.)
3. Adjusted " to run in testsuite/, like setup.sh

infispiel committed:
Run tests on all dev branches

infispiel commented:
@alvacouch Please take a look at the changes before I merge.
@alexisylee Please look at commits [d26b7ad](https://github.com/TuftsBCB/medford/pull/25/commits/d26b7ad40103516f34a14ee3b5169f40220be995) , [1e3ad6f](https://github.com/TuftsBCB/medford/pull/25/commits/1e3ad6fd15a4dd7b9bcf94a350e6ae7756d4b692) and make sure you understand the changes that I did. Also, moving forwards, please make sure to make your branches off of the dev branch.
